components/esp_websocket_client/include/esp_websocket_client.h

```diff
@@ -166,6 +166,16 @@ esp_err_t esp_websocket_client_stop(esp_websocket_client_handle_t client);
  */
 esp_err_t esp_websocket_client_destroy(esp_websocket_client_handle_t client);
 
+/**
+ * @brief      If `selected` set to True, WebSocket client will destroy and free all resources at the end of event loop.
+ *
+ * @param[in]  client      The client
+ * @param[in]  selected    The flag
+ *
+ * @return     esp_err_t
+ */
+esp_err_t esp_websocket_client_destroy_selected(esp_websocket_client_handle_t client, bool selected);
```

Comment:
Does it make sense to add the `bool selected` parameter? Why would we call: `esp_websocket_client_destroy_selected(..., false)` ?

Reply:
@david-cermak not really sure about such scenario just wanted to let the users freedom to change the flag in case if it needed in some case. But without `bool selected` parameter it will be more convenient from API perspective and this part https://github.com/espressif/esp-protocols/pull/211#discussion_r1084946518 part can be reused in that case. So will modify to make it without parameter. Thanks!